### ISSUE TITLE
Appium support using Howitzer POM

### ISSUE THREAD

**Issue by jaishree-patidar** [strongqa/howitzer#265]
## Meta
Howitzer Version:
2.0.3
Appium Version: 1.7.1

Driver and browser information:
```
Appium_capybara 1.4.1
capybara-2.16.1
selenium-webdriver-3.8.0
```

Gem Versions:
```
gem 'selenium-webdriver', '~> 3.7'
gem 'capybara', '~> 2.16', '>= 2.16.1'
gem 'appium_capybara', '~> 1.4'
```

## Expected Behavior
Scripts created for selenium desktop browser should work on Appium iOS device safari by updating the browser creation.

## Actual Behavior
On using Howitzer POM, I am getting error on instance creation [Error text:Method has not yet been implemented (Selenium::WebDriver::Error::UnknownError)] . 
Whereas, I am able to automate a script using SitePrism.

Appium Logs : [appium-server-logs.txt](https://github.com/strongqa/howitzer/files/1605741/appium-server-logs.txt)

**Safari gets launched on simulator and navigates to URL but getting error on given
Error text:Method has not yet been implemented (Selenium::WebDriver::Error::UnknownError)**

## Steps to reproduce
1. Creation of Appium driver with Capybara

```
# :appium driver
Capybara.register_driver(:appium) do |app|
  desired_caps_ios = {
      browserName: "Safari",
      deviceName: "iPad Air 2",
      platformName: "iOS",
      udid: "1FC2F6A8-ACB0-4F8D-94C8-F4242AE429E9",
      automationName: "XCUITest",
      autoWebview: true
  }
  url = "http://127.0.0.1:4723/wd/hub" # or a sauce labs url

  appium_lib_options = {
      server_url: url
  }
  all_options = {
      appium_lib: appium_lib_options,
      caps: desired_caps_ios
  }
  Appium::Capybara::Driver.new app, all_options
end
Capybara.default_driver = :appium
```

2. Page code - 

**With Howitzer -- Not Working**
```
class LandingPage < Howitzer::Web::Page
  path '/'

  validate :element_presence, :main_container
  element :main_container, '.main-container'
  element :get_started, 'button'

  def clickGetStarted
    get_started_element.click
  end
end
```

**With SitePrism -- Working**
```
module Pages
  class Home < SitePrism::Page
    element :get_started, 'Button'

    def click_get_started
      get_started.click
    end
  end
end
```

3. Cucumber feature step definition

```
And(/^I click Get Started on the Home Page$/) do
  LandingPage.given.clickGetStarted  #-- giving error on given
end
```

**Comment by Lec1k** [comment]
It worked fine for me:
```ruby
Capybara.register_driver(:appium) do |app|
  caps = {}
  caps['appiumVersion'] = '1.7.2'
  caps['deviceName'] = 'iPad Air 2 Simulator'
  caps['deviceOrientation'] = 'portrait'
  caps['platformVersion'] = '10.0'
  caps['platformName'] = 'iOS'
  caps['browserName'] = 'Safari'

  url = "" # sauce labs url

  appium_lib_options = {
      server_url: url
  }
  all_options = {
      appium_lib: appium_lib_options,
      caps: caps
  }
  Appium::Capybara::Driver.new app, all_options
end
```

Page code:
```ruby
class HomePage < Howitzer::Web::Page
  path '/'
  validate :title, /\ADemo web application - Home\z/
  validate :url, %r{\A(?:.*?:\/\/)?[^\/]*\/?\z}
  validate :element_presence, :navigation_home

  element :article_link, :xpath, ->(title) { ".//a[.='#{title}']" }
  element :panel_heading, :xpath, ->(title) { "//h3[text()=\"#{title}\"]/parent::*/following-sibling::*" }
  element :navigation_home, 'a.navbar-brand'


  def click_navigation_home
    navigation_home_element.click
  end
end
```
cucumber steps:
    When I open home page
    And I click home at navigation panel
    Then I should be redirected to home page

```ruby
When /^I open (.+?) page$/ do |page|
  page.open
end

When 'I click home at navigation panel' do
  HomePage.on { click_navigation_home }
end

Then /I should be redirected to (.+) page/ do |page|
  expect(page).to be_displayed
end
```

It might be caused by wrong element locator,  in howitzer example you wrote: 
`element :get_started, 'button'` , but in site_prism example: `element :get_started, 'Button'
`

**Comment by jaishree-patidar** [comment]
@Lec1k 
Sorry for the typo in the example I shared in the issue. Wrong element locator is not the issue as I have tried various locators on different pages. I am still getting this error. Might be something with the Appium version I am using. I am using Appium 1.7.1 whereas you tried on 1.7.2. I will update my Appium version and give it a try.

Also, it would be helpful if you can share the versions of gems you have used.

**Comment by jaishree-patidar** [comment]
@Lec1k 
I updated my Appium version to 1.7.2 and also tried changing calling of function from "LandingPage.given.functionName" to "LandingPage.on { functionName }" but no luck.

May be I am missing out something. It would be really helpful if you can share the versions of gems you have used.

**Comment by Lec1k** [comment]
@jaishree-patidar 
```
Gems included by the bundle:
  * activemodel (5.1.1)
  * activesupport (5.1.1)
  * addressable (2.5.2)
  * appium_capybara (1.4.1)
  * appium_lib (9.8.2)
  * appium_lib_core (1.2.4)
  * ast (2.3.0)
  * backports (3.10.3)
  * builder (3.2.3)
  * bundler (1.15.1)
  * byebug (9.1.0)
  * capybara (2.13.0)
  * capybara-screenshot (1.0.18)
  * capybara-webkit (1.14.0)
  * childprocess (0.8.0)
  * cliver (0.3.2)
  * coderay (1.1.2)
  * colorize (0.8.1)
  * concurrent-ruby (1.0.5)
  * cucumber (3.0.2)
  * cucumber-core (3.0.0)
  * cucumber-expressions (4.0.4)
  * cucumber-tag_expressions (1.0.1)
  * cucumber-wire (0.0.1)
  * diff-lcs (1.3)
  * domain_name (0.5.20170404)
  * factory_bot (4.8.2)
  * faraday (0.12.1)
  * faraday_middleware (0.11.0.1)
  * ffi (1.9.18)
  * gherkin (4.1.3)
  * gli (2.17.1)
  * gmail (0.6.0)
  * gmail_xoauth (0.4.2)
  * headless (2.3.1)
  * howitzer (2.1.1)
  * http-cookie (1.0.3)
  * i18n (0.9.1)
  * json (2.1.0)
  * launchy (2.4.3)
  * log4r (1.1.10)
  * mail (2.7.0)
  * method_source (0.9.0)
  * mime-types (3.1)
  * mime-types-data (3.2016.0521)
  * mini_mime (1.0.0)
  * mini_portile2 (2.3.0)
  * minitest (5.10.3)
  * multi_json (1.12.2)
  * multi_test (0.1.2)
  * multipart-post (2.0.0)
  * netrc (0.11.0)
  * nokogiri (1.8.1)
  * oauth (0.5.3)
  * parallel (1.12.0)
  * parser (2.4.0.2)
  * poltergeist (1.14.0 f7a8e97)
  * powerpack (0.1.1)
  * pry (0.11.3)
  * pry-byebug (3.5.0)
  * public_suffix (3.0.1)
  * rack (2.0.3)
  * rack-test (0.8.0)
  * rainbow (2.2.2)
  * rake (12.3.0)
  * repeater (0.0.1)
  * rest-client (2.0.2)
  * rspec (3.7.0)
  * rspec-core (3.7.0)
  * rspec-expectations (3.7.0)
  * rspec-mocks (3.7.0)
  * rspec-support (3.7.0)
  * rspec-wait (0.0.9)
  * rubocop (0.51.0)
  * ruby-progressbar (1.9.0)
  * rubyzip (1.2.1)
  * selenium-webdriver (3.7.0)
  * sexy_settings (0.1.0)
  * spyke (5.0.0)
  * syntax (1.2.2)
  * thread_safe (0.3.6)
  * tomlrb (1.2.6)
  * tzinfo (1.2.4)
  * unf (0.1.4)
  * unf_ext (0.0.7.4)
  * unicode-display_width (1.3.0)
  * uri_template (0.7.0)
  * websocket-driver (0.6.5)
  * websocket-extensions (0.1.2)
  * xpath (2.1.0)
```
Can you share the url of the application you are testing? I can try your code on my instance of howitzer.

**Comment by romikoops** [comment]
In master. Will be released in v2.1.2
